#! /bin/sh /usr/share/dpatch/dpatch-run
## 11_check_disk_smb_NT_STATUS_ACCESS_DENIED.dpatch by
## Matthias Eble <psychotrahe@users.sourceforge.net>
##
## From 3c67c9cb60a4681a29e509fccbb333e0f6234f54 Mon Sep 17 00:00:00 2001
## From: Matthias Eble <psychotrahe@users.sourceforge.net>
## Date: Sun, 28 Nov 2010 21:43:48 +0100
## Subject: [PATCH] check_disk_smb now handles NT_STATUS_ACCESS_DENIED properly
## (Debian #601696)
##
## DP: Handles NT_STATUS_ACCESS_DENIED properly (http://bugs.debian.org/601696)

@DPATCH@

diff --git a/plugins-scripts/check_disk_smb.pl b/plugins-scripts/check_disk_smb.pl
index 4698700..6783543 100755
--- a/plugins-scripts/check_disk_smb.pl
+++ b/plugins-scripts/check_disk_smb.pl
@@ -239,7 +239,7 @@ if (/\s*(\d*) blocks of size (\d*)\. (\d*) blocks available/) {
 	$answer = "Result from smbclient not suitable\n";
 	$state = "UNKNOWN";
 	foreach (@lines) {
-		if (/(Access denied|NT_STATUS_LOGON_FAILURE)/) {
+		if (/(Access denied|NT_STATUS_LOGON_FAILURE|NT_STATUS_ACCESS_DENIED)/) {
 			$answer = "Access Denied\n";
 			$state = "CRITICAL";
 			last;

